packages/graphql-server-core/src/runHttpQuery.ts

```diff
@@ -68,6 +70,13 @@ export async function runHttpQuery(handlerArguments: Array<any>, request: HttpQu
 
   let responses: Array<ExecutionResult> = [];
   for (let requestParams of requestPayload) {
+    if ( isGetRequest && !requestParams.query.trim().startsWith('query')) {
```

Comment:
Also, the query isn't necessarily the first thing in the document right? For example:

```
fragment X on Y { ... }

{ ... }
```

Is also a valid query.